### PR TITLE
[T2A5][W09-A3]Xiao Pu

### PR DESCRIPTION
**Pull request by xpdavid**

### CODE REVIEW

**Comment by xpdavid** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -130,21 +133,22 @@
...
+     * For example, a person's name is stored in the hashmap with key `name`,
+     * we use enum type to store name/email/phone
      */
-    private static final int PERSON_DATA_INDEX_NAME = 0;
-    private static final int PERSON_DATA_INDEX_PHONE = 1;
-    private static final int PERSON_DATA_INDEX_EMAIL = 2;
-
+    private enum PersonProperty {
+        NAME, EMAIL, PHONE;
+    };
```
Comment (z11i) [inline src/seedu/addressbook/AddressBook.java:144]: Enum :+1:

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -157,95 +161,86 @@
...
+    /**
+     * The main method will automatically run when the programme get executed.
+     * 
+     * @param args   will use filepath in args[0] to store data rather than use default file to store it. 
      */
     public static void main(String[] args) {
         showWelcomeMessage();
         processProgramArgs(args);
         loadDataFromStorage();
         while (true) {
```
Comment (z11i) [inline src/seedu/addressbook/AddressBook.java:200]: Should we put the `while` loop here, or in the extracted method?

**Comment by z11i** [comment]
Very nicely done! Good job!

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -157,95 +161,86 @@
...
     }
 
-    /*
-     * ==============NOTE TO STUDENTS======================================
-     * The method header comment can be omitted if the method is trivial
-     * and the header comment is going to be almost identical to the method
-     * signature anyway.
-     * ====================================================================
+    /**
+     * Get user input, process it, and output command result
```
Comment (damithc) [inline src/seedu/addressbook/AddressBook.java:206]: `Gets ...`

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -157,95 +161,86 @@
...
+     */
+    private static void handleUserCommand() {
+        String userCommand = getUserInput();
+        echoUserCommand(userCommand);
+        String feedback = executeCommand(userCommand);
+        showResultToUser(feedback);
+    }
+
+    /**
+     * The method will show the welcome message to user
```
Comment (damithc) [inline src/seedu/addressbook/AddressBook.java:216]: Based on the header comment convention followed in other methods, this should be: `Shows the welcome ...`

**Comment by damithc** [comment]
Good work putting in a lot of effort @xpdavid 
Some comments added. When you add features, don't forget to update user guide.

**Pull request closed without merging**